src/hash.cr

```diff
@@ -272,6 +272,26 @@ class Hash(K, V)
     self
   end
 
+  def select(&block : K, V -> U)
+    inject({} of K => V) do |memo, k, v|
+      yield(k, v) ? memo.merge!({k => v}) : memo
+    end
+  end
+
+  def select!(&block : K, V -> U)
+    delete_if{|k, v| !yield k, v }
```

Comment:
1. *space before `{`* where? in `delete_if{`? I don't like space there
2. Never ever base any style-guide on bbatsov thing, because that project was mental from the beginning -- full of random nonsense, that influenced others a lot, because more clever guys were not ready to tell others how to write code, but these illogical amateurs were.